src/libasr/runtime/lfortran_intrinsics.c

```diff
@@ -113,17 +113,297 @@ LFORTRAN_API void _lfortran_printf(const char* format, ...)
     va_end(args);
 }
 
+char* substring(const char* str, int start, int end) {
+    int len = end - start;
+    char* substr = (char*)malloc((len + 1) * sizeof(char));
+    strncpy(substr, str + start, len);
+    substr[len] = '\0';
+    return substr;
+}
+
+char* appendToString(char* str, const char* append) {
```

Comment:
Let's use this convention, consistent with the rest of the file:
```suggestion
char* append_to_string(char* str, const char* append) {
```